src/generic/UnivPoly.jl

```diff
@@ -620,15 +616,7 @@ function isless(a::UnivPoly{T}, b::UnivPoly{T}) where {T}
    end
    S = parent(a)
    num = nvars(S)
-   s = data(a)
-   t = data(b)
-   if nvars(parent(s)) != num
-      s = upgrade(S, s)
-   end
-   if nvars(parent(t)) != num
-      t = upgrade(S, t)
-   end
-   return isless(s, t)
+   return isless(upgrade(S, data(a)), upgrade(S, data(b)))
```

Comment:
So we upgrade the underlying polynomials of `a` and `b` and then throw this away... I always wondered if that's a good idea. Well, it is certainly wise not to change this in this PR, but perhaps in a future PR we could think about introducing another helper (distinct from the one I suggest in another comment) like this:
```julia
function upgrade!(p::UniversalPolyRingElem)
  p.p = upgrade(parent(p), data(p))
  return p
end
```
and then here we can do
```suggestion
   return isless(upgrade!(a), upgrade!(b))
```

Reply:
Yes, I thought about this too. I wasn't sure if it is a good idea that `isless` potentially modifies its inputs. Of course they would only be modified internally but shouldn't `isless` be named `isless!` then? (I'm not actually suggesting to rename `isless`, just wondering.)

Reply:
Could you put this, and also the change that upgrade checks if it needs to do anything, into its own PR to not clutter this one here with mostly unrelated changes?

Reply:
In which order would you like to have the changes?